mxfusion/components/distributions/gp/cond_gp.py

```diff
@@ -232,3 +236,61 @@ def replicate_self(self, attribute_map=None):
         replicant._has_mean_cond = self._has_mean_cond
         replicant.kernel = self.kernel.replicate_self(attribute_map)
         return replicant
+
+    def marginalise_conditional_distribution(self, X, X_cond, q_u, n_outputs, jitter, diagonal_variance,
+                                             **kernel_params):
+        """
```

Comment:
Please provide more details in the docstring what is the definition of the arguments that go in the function. In particular, what is the expected dimensionality.